crates/nu-path/src/helpers.rs

```diff
@@ -1,13 +1,41 @@
+use etcetera::BaseStrategy;
 #[cfg(windows)]
 use omnipath::WinPathExt;
+use once_cell::sync::Lazy;
 use std::path::PathBuf;
 
+static HOME_DIR: Lazy<Option<PathBuf>> = Lazy::new(|| etcetera::home_dir().ok());
+static CONFIG_DIR: Lazy<Option<PathBuf>> = Lazy::new(|| {
+    let default = if let Some(mut config_dir) = home_dir() {
```

Comment:
Please correct me if I'm wrong, but this isn't doing what we want. The way I read it, the defaults are set like this.
```
HOME_DIR
apple: HOME env var
linux: HOME env var
windows: USERPROFILE env var

CONFIG_DIR default
apple: HOME_DIR/.config/nushell if HOME_DIR is set, otherwise None
linux: HOME_DIR/.config/nushell if HOME_DIR is set, otherwise None
windows: USERPROFILE/AppData/Roaming/nushell
```
I believe we want the default, across all operating systems to be `HOME_DIR/.config/nushell`

Reply:
The more I look at what I wrote, the more I think that I may have the windows path wrong. Another reason why comments explicitly stating what they are would be helpful.

Reply:
There are 3 cases:
- `~/.config/nushell` exists -> use that
- `"native"/nushell` exists -> use that
- neither exists, return `~/.config/nushell`

We need to return either `~/.config` or `"native"`. The path shouldn't contain `nushell` part.

Here, "native" means `~/.config` on Linux, `~/Library/Application Support` on macOS, & `~\AppData\Roaming` on Windows.

Reply:
ok, I'm going to try this soon. Thanks for explaining.